### PR TITLE
Add `vcat` implementation for `Data` trait

### LINKED ISSUE #519: Implement vcat for Data trait
It is very common to concatenate the "rows" of geospatial data into a single data set. DataFrames.jl currently provides an implementation that works just for DataFrame. We need an extension that works with Data using the Tables.jl interface already in place.

### PR DESCRIPTION
**Pull request by eliascarv**
closes #519

### CODE REVIEW

File: src/data.jl [hunk 1]
```diff
@@ -224,6 +224,34 @@ function Base.getindex(data::Data, inds, var::Regex)
...
+  dom1 = domain(data1)
+  dom2 = domain(data2)
+
+  cols1 = Tables.columns(tab1)
+  cols2 = Tables.columns(tab2)
+  names = Tables.columnnames(cols1)
+  
+  if Set(names) ≠ Set(Tables.columnnames(cols2))
+    throw(ArgumentError("All data must have the same variables"))
+  end
```
Comment (juliohm) [inline src/data.jl:241]: This is not necessarily the case. As we discussed before, the DataFrames.jl package supports the concatenation even when some columns are not present via the option cols=:union (if I remember correctly).

File: src/data.jl [hunk 2]
```diff
@@ -224,6 +224,34 @@ function Base.getindex(data::Data, inds, var::Regex)
...
+
+  columns = map(names) do name
+    column1 = Tables.getcolumn(cols1, name)
+    column2 = Tables.getcolumn(cols2, name)
+    vcat(column1, column2)
```
Comment (juliohm) [inline src/data.jl]:
```suggestion
    [column1; column2]
```

File: src/data.jl [hunk 3]
```diff
@@ -224,6 +224,34 @@ function Base.getindex(data::Data, inds, var::Regex)
...
+  end
+
+  newtab = (; zip(names, columns)...)
+  newdom = GeometrySet(vcat(collect(dom1), collect(dom2)))
```
Comment (juliohm) [inline src/data.jl]:
```suggestion
  newdom = GeometrySet([collect(dom1), collect(dom2)])
```

File: src/data.jl (hunk 1 above)
Comment (eliascarv) [reply src/data.jl:241]: I remember this. But my intention was to start with a simpler implementation. Then we can add more features in future PRs.

Comment (juliohm) [reply src/data.jl:241]: Makes total sense. Good strategy, just need to communicate better the plan.

**Review by juliohm — APPROVED** [review]

File: src/data.jl [hunk 4]
```diff
@@ -224,6 +224,34 @@ function Base.getindex(data::Data, inds, var::Regex)
...
+  end
+
+  columns = map(names) do name
+    column1 = Tables.getcolumn(cols1, name)
+    column2 = Tables.getcolumn(cols2, name)
+    [column1; column2]
+  end
+
+  newtab = (; zip(names, columns)...)
+  newdom = GeometrySet([collect(dom1), collect(dom2)])
```
Comment (eliascarv) [inline src/data.jl]:
```suggestion
  newdom = GeometrySet([collect(dom1); collect(dom2)])
```

**Pull request merged**